wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/model/impl/ProfileImpl.java

```diff
@@ -21,19 +31,36 @@
  *
  * @author Yevhenii Voevodin
  */
+@Entity(name = "Profile")
 public class ProfileImpl implements Profile {
 
-    private String              id;
-    private String              email;
+    @Id
+    @Column(name = "user_id")
+    private String userId;
+
+    @PrimaryKeyJoinColumn
+    public UserImpl user;
+
+    @ElementCollection
+    @MapKeyColumn(name = "name")
+    @Column(name = "value")
+    @CollectionTable(joinColumns = @JoinColumn(name = "user_id"),
+                     uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "name"}))
     private Map<String, String> attributes;
 
-    public ProfileImpl(String id, String email) {
-        this.id = id;
+    // jpa mapping delegated to 'user' field
+    @Transient
```

Comment:
in which case do we need to create a ProfileImpl directly with a userId and a given email instead of getting email through the user object ?

If Profile has a JPA relationship with User object, there is no need to store/wrap email in any case as you can do getProfile().getUser().getEmail() as well